extensions/WSLExtension/DevHomeProviders/WslProvider.cs

```diff
@@ -97,4 +109,31 @@ public IAsyncOperation<ComputeSystemsResult> GetComputeSystemsAsync(IDeveloperId
         var notImplementedException = new NotImplementedException($"Method not implemented by WSL Compute System Provider");
         return new ComputeSystemAdaptiveCardResult(notImplementedException, notImplementedException.Message, notImplementedException.Message);
     }
+
+    /// <summary>
+    /// Gets the localized error message for when the virtual machine platform Windows optional component
+    /// is not enabled.
+    /// </summary>
+    /// <returns>
+    /// Returns the error message for when the feature is not enabled and returns an empty string when the feature is enabled.
+    /// </returns>
+    private string GetErrorStringForWhenFeatureNotEnabled()

```

Comment:
I think this name/the way it works is a little confusing -- I'd expect it to always get a string, and a different check for enablement. Can we have a check that returns a bool and then get the string if we need to?

Reply:
fair, I figured I wanted to do it all in one method. But if you think its confusing, I can split it in two.

Reply:
I can go either way.  But separating out the calls will reduce the number of negations in the name.

    var isSupported = IsBlahSupported();
    if(!isSupported)
    {
        var errorString = GetErrorString();
    }

something like that.